parser/Builder.cc

```diff
@@ -148,6 +148,41 @@ class Builder::Impl {
             auto name = id->name.data(gs_);
             ENFORCE(name->kind == core::NameKind::UTF8);
             auto name_str = name->show(gs_);
+            if (isNumberedParameterName(name_str) && driver_->lex.context.inDynamicBlock()) {
+                if (driver_->numparam_stack.has_ordinary_params()) {
+                    error(ruby_parser::dclass::OrdinaryParamDefined, id->loc);
+                }
+
+                auto raw_context = driver_->lex.context.stackCopy();
+                auto raw_numparam_stack = driver_->numparam_stack.stackCopy();
```

Comment:
Why did we make a separate method just to copy vectors?

This is the same as using the copy assignment operator:

```suggestion
                auto raw_context = driver_->lex.context;
                auto raw_numparam_stack = driver_->numparam_stack;
```

(and then you should be able to delete the `stackCopy` function defined elsewhere.

Reply:
In both case we're not copying the context itself but the inner stack which is private so we need a public method to act as an accessor.

I simplified the bodies to use your suggestion tho.